src/schema/rules/sidecars/mrs.yaml

```
@@ -0,0 +1,98 @@
+#
+# Groups of related metadata fields
+#
+# Assumptions: never need disjunction of selectors
+# Assumptions: top-to-bottom overrides is sufficient logic
+
+---
+# MRS Common metadata fields
+MRSScannerHardware:
+  selectors:
+    - modality == "mrs"
+  fields:
+    InstitutionName: recommended
+    InstitutionAddress: recommended
+    InstitutionalDepartmentName: recommended
+    Manufacturer: recommended
+    ManufacturersModelName: recommended
+    DeviceSerialNumber: recommended
+    StationName: recommended
+    SoftwareVersions: recommended
+    MagneticFieldStrength: recommended
+    ReceiveCoilName: recommended
+    ReceiveCoilActiveElements: recommended
+    NumberReceiveCoilActiveElements: optional
+    NumberTransmitCoilActiveElements: optional
+
+MRSSample:
+  selectors:
+    - modality == "mrs"
+  fields:
+    BodyPart:
+      level: optional
+      description_addendum: Corresponds to DICOM Tag 0018, 0015 `Body Part Examined`.
+    BodyPartDetails: optional
+    BodyPartDetailsOntology: optional
+
+MRSSequenceSpecifics:
+  selectors:
+    - modality == "mrs"
+  fields:
+    PulseSequenceType:
+      level: recommended
+      description_addendum: |
+        For example, `"sLASER"`, `"MEGA-PRESS"`, `"EPSI"`, `"Metabolite-cycled MRSI"`.
+    ScanningSequence__mrs: recommended
+    SequenceName: recommended
+    PulseSequenceDetails: recommended
+    MRAcquisitionType: recommended for MRSI
+    WaterSuppression: recommended
+    WaterSuppressionTechnique: optional
+    OuterVolumeSuppression: recommended
+    B0ShimmingTechnique: optional
+    B1ShimmingTechnique: optional
+
+MRSRequiredFields:
+  selectors:
+    - modality == "mrs"
+  fields:
+    ResonantNucleus: required
+    SpectrometerFrequency: required
+    SpectralWidth: required
+    EchoTime: required
+
+MRSRecommendedFields:
+  selectors:
+    - modality == "mrs"
+  fields:
+    NumberOfSpectralPoints: recommended
+    NumberOfTransients:
+      level: optional
+      level_addendum: recommended for SVS and unloc
+    RepetitionTime__mrs: recommended
+    MixingTime: recommended
+    InversionTime: recommended for inversion recovery data
+    FlipAngle: recommended
+    AcquisitionVoxelSize: recommended
+    MatrixSize: recommended for MRSI
+    VolumeAffineMatrix: recommended for MRSI
+    EditTarget: recommended for edited data
+    EditPulse: recommended for edited data
+    EditCondition: recommended for edited data
+    EncodingTechnique: recommended for MRSI
+    ReferenceSignal: recommended if MRS reference data are present
+    AnatomicalImage: recommended if anatomical MRI data are present
```

This is a fun one:

```suggestion
    AnatomicalImage: optional, but recommended if anatomical MRI data are present
```

\+

```yaml
MRSAnatomicalImageRec:
  selectors:
    - modality == "mrs"
    - '"anat" in dataset.datatypes'
  fields:
    AnatomicalImage: recommended
```

---

I'm not sure how that suggested change and that extra code work together with respect to what appears in `src/modality-specific-files/magnetic-resonance-spectroscopy.html`. That is, I've included:

```
MRSRecommendedFieldsAnatomicalImage:
  selectors:
    - modality == "mrs"
    - '"anat" in dataset.datatypes'
  fields:
    AnatomicalImage: recommended
```

and:

```
{{ MACROS___make_sidecar_table([
  "mrs.MRSRecommendedFields",
  "mrs.MRSRecommendedFieldsNumTransients",
  "mrs.MRSRecommendedFieldsAnatomicalImage",
  "mrs.MRSIRecommendedFields"
  ])
}}
```

But now `AnatomicalImage` is shown as `RECOMMENDED` rather than `OPTIONAL, but RECOMMENDED if anatomical MRI data are present`.